corehq/apps/users/forms.py

```diff
@@ -869,6 +869,15 @@ def clean_assigned_locations(self):
 
         return [location.location_id for location in locations]
 
+    def clean(self):
+        cleaned_data = super(CommtrackUserForm, self).clean()
+
+        primary_location_id = cleaned_data['primary_location']
+        assigned_location_ids = cleaned_data['assigned_locations']
+        if primary_location_id:
+            if primary_location_id not in assigned_location_ids:
+                raise ValidationError(_("Only one of the user's locations can be a primary location"))
```

Comment:
You can attach the error to a specific field, though.  I wasn't actually sure if that's possible, but it looks like Django supports that in this form:
```python
self.add_error('primary_location',
               _("Only one of the user's locations can be a primary location"))
```
https://docs.djangoproject.com/en/1.10/ref/forms/api/#django.forms.Form.add_error